vendor/bat-native-ads/src/bat/ads/internal/ads/search_result_ad.cc

```diff
@@ -30,18 +32,45 @@ SearchResultAd::~SearchResultAd() {
 
 void SearchResultAd::TriggerEvent(
     mojom::SearchResultAdInfoPtr ad_mojom,
-    const mojom::SearchResultAdEventType event_type,
-    TriggerSearchResultAdEventCallback callback) {
+    const mojom::SearchResultAdEventType event_type) {
+  if (event_type == mojom::SearchResultAdEventType::kViewed) {
+    ad_viewed_event_queue_.push_front(std::move(ad_mojom));
+    MaybeTriggerAdViewedEventFromQueue();
+    return;
+  }
+
   event_handler_->FireEvent(
       ad_mojom, event_type,
       [=](const bool success, const std::string& placement_id,
-          const mojom::SearchResultAdEventType event_type) {
-        callback(success, placement_id, event_type);
-      });
+          const mojom::SearchResultAdEventType event_type) {});
```

Comment:
seems strange doing nothing here.

Reply:
yeah, we have nothing to do in case of clicked event

Reply:
Added comment